commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mcollovati.quarkus.hilla.deployment;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Scanner;
+
+import com.vaadin.hilla.Endpoint;
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.deployment.CodeGenContext;
+import io.quarkus.deployment.CodeGenProvider;
+import org.eclipse.microprofile.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TypescriptClientCodeGenProvider implements CodeGenProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TypescriptClientCodeGenProvider.class);
+
+    static final String FRONTEND_FOLDER_NAME = "frontend";
+    private Path frontendFolder;
+
+    @Override
+    public String providerId() {
+        return "quarkus-hilla-connect-client";
+    }
+
+    @Override
+    public String inputDirectory() {
+        return FRONTEND_FOLDER_NAME;
+    }
+
+    @Override
+    public void init(ApplicationModel model, Map<String, String> properties) {
+        if (model.getApplicationModule() == null) {
+            LOGGER.info("ApplicationModule is null");
+            return;
+        }
+        File moduleDir = model.getApplicationModule().getModuleDir();
+        Path legacyFrontendFolder = moduleDir.toPath().resolve(FRONTEND_FOLDER_NAME);
+        Path frontendDir = moduleDir.toPath().resolve(Path.of("src", "main", FRONTEND_FOLDER_NAME));
+        if (Files.isDirectory(frontendDir)) {
+            this.frontendFolder = frontendDir;
+            LOGGER.debug("Using frontend folder {}", frontendDir);
+        } else if (Files.isDirectory(legacyFrontendFolder)) {
+            this.frontendFolder = legacyFrontendFolder;
+            LOGGER.debug("Using legacy frontend folder {}", legacyFrontendFolder);
+        } else {
+            LOGGER.debug("Frontend folder not detected");
+        }
+    }
+
+    @Override
+    public Path getInputDirectory() {
+        return frontendFolder;
+    }
+
+    @Override
+    public boolean shouldRun(Path sourceDir, Config config) {
+        if (!Files.isDirectory(sourceDir)) {
+            return false;
+        }
+        String prefix = computeConnectClientPrefix(config);
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = sourceDir.resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            try {
+                String content = Files.readString(customClient);
+                if (!content.contains("const client = new ConnectClient({prefix: '" + prefix + "'});")) {
```

Comment:
Use the regex from later to be a bit more flexible?
I am not sure if my assumption is correct, that the user could also modify the connect client and it gets patched. Or do we assume always a quarkus-hilla generated one ?

Reply:
The idea is to patch the file only if the client creation statement matches the template

Reply:
Yea thats good. Idea was just to be more unrestrictive, but the question is, in how many cases that would happen.